File: typical90/10.cpp
#include <bits/stdc++.h>
using namespace std;
/* alias */
using ll = long long;
using pii = pair<int, int>;
using pll = pair<ll, ll>;
using vi = vector<int>;
using vl = vector<ll>;
using vvl = vector<vector<ll>>;
using vs = vector<string>;
using Graph = vector<vector<ll>>;

const ll MOD = 1000000007;
const ll INF = 2000000000;
/* define short */
#define pb push_back
#define mp make_pair

/* repeat macro */
#define rep(i, n) for (ll i = 0; i < (ll)n; i++)
#define rep2(i, s, n) for (ll i = s; i < (ll)n; i++)

int main() {
    ll n;
    cin >> n;
    vl c(n), p(n);
    rep(i, n) cin >> c[i] >> p[i];
    ll q;
    cin >> q;
    vl l(q), r(q);
    rep(i, q) cin >> l[i] >> r[i];
    rep(i, q) {
        l[i]--;
        r[i]--;
    }
    vl acc1(n + 1), acc2(n + 1);
    if (c[0] == 1)
        acc1[1] = p[0];
    else
        acc2[1] = p[0];
    rep2(i, 2, n + 1) {
        if (c[i] == 1) {
            acc1[i] = acc1[i - 1] + p[i];
            acc2[i] = acc2[i - 1];
        } else {
            acc1[i] = acc1[i - 1];
            acc2[i] = acc2[i - 1] + p[i];
        }
    }
    rep2(i, 1, q + 1) {
        ll a, b;
        a = acc1[r[i]] - acc1[l[i] - 1];
        b = acc2[r[i]] - acc2[l[i] - 1];
        cout << a << " " << b << endl;
    }
    // rep2(i, 1, n + 1) { cout << acc1[i] << " " << acc2[i] << endl; }
}

File: typical90/10.py
# input
n = int(input())
c = []
p = []
for _ in range(n):
    ci, pi = map(int, input().split())
    c.append(ci)
    p.append(pi)
q = int(input())
l = []
r = []
for _ in range(q):
    li, ri = map(int, input().split())
    li -= 1
    ri -= 1
    l.append(li)
    r.append(ri)
# main
acc1 = [0 for _ in range(n)]
acc2 = [0 for _ in range(n)]
acc1[0] = p[0] if c[0] == 1 else 0
acc2[0] = p[0] if c[0] == 2 else 0

for i in range(1, n):
    if c[i] == 1:
        acc1[i] += acc1[i - 1] + p[i]
        acc2[i] = acc2[i - 1]
    else:
        acc1[i] = acc1[i - 1]
        acc2[i] += acc2[i - 1] + p[i]

for li, ri in zip(l, r):
    a = acc1[ri] - acc1[li] + (p[li] if c[li] == 1 else 0)
    b = acc2[ri] - acc2[li] + (p[li] if c[li] == 2 else 0)
    print(a, b)

